src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -254,7 +274,25 @@ private Result executeCmd(String cmd) throws IOException, InterruptedException {
                     .stderr(stderr)
                     .cmdAsSingleString(cmd)
                     .start().join() == 0;
-            String stdoutStr = stdout.toString();
+
+            // capture the stdout so it can be parsed later on
+            String stdoutStr = null;
+            try {
+                Computer computer = Computer.currentComputer();
+                if (computer != null) {
+                    Charset charset = computer.getDefaultCharset();
+                    if (charset != null) {
+                        baos.flush();
```

Comment:
Ahh no, I think you misinterpret.

The OS will buffer what it sends to the stream of the reading process, chunking typically per line or unless the invoking process does a read. As we are using an output stream rather than pulling from the input until EOF we cannot be certain that we have reached the EOF and thus cannot be certain that `baos` has the full contents.

Thus the call to flush in this case is a no-op.